draft-thoji-scone-trone-protocol.md

```diff
@@ -304,18 +338,78 @@ The following pseudocode indicates how a network element might detect a TRONE
 packet and replace an existing rate signal.
 
 ~~~ pseudocode
+target_rate_value = convert_rate_to_signal(target_rate, version)
+
 is_long = packet[0] & 0x80 == 0x80
-is_trone = compare(packet[1..5], TRONE_VERSION)
-if is_long and is_trone:
-  dcid_len = packet[5]
-  offset = 6 + dcid_len
-  scid_len = packet[offset]
-  offset = offset + 1 + scid_len
-  packet_rate = read_rate_signal(packet[offset : offset + rate_signal_size])
-  if target_rate.is_lower_than(packet_rate):
-    write_rate_signal(packet[offset : offset + rate_signal_size], target_rate)
+is_trone1 = compare(packet[1..5], TRONE1_VERSION)
+is_trone2 = compare(packet[1..5], TRONE2_VERSION)
+
+if is_long and (is_trone1 or is_trone2):
+  packet_rate_value = packet[0] & 0x3f
+  if packet_rate_value == 0x3f or target_rate_value < packet_rate_value:
+    packet[0] = packet[0] & 0xc0 | target_rate_value
 ~~~
 
+For TRONE packets with two versions, a network element applies the same rate
+policy to both, accounting for the different scales of each version. When a
+network element signals a rate limit that falls within the range of both
+versions, it should signal appropriate values in each version. When the target
+rate limit is outside the range of a particular version, the network element
+should signal the minimum value (0) for rates below the range and preserve the
+"no limit" value (63) for rates above the range.
```

Comment:
```suggestion
should signal the minimum value (0) for rates below the range and preserve the
original value for rates above the range.
```
When there are multiple network elements, the Rate Signal field might convey a value other than 63.